test/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -3523,12 +3563,262 @@ public void earlyStepOutNonIncremental(boolean constructFromBytes) throws Except
+    @CsvSource({
+        "true, 5E",
+        "false, 5F",
+    })
+    public void readBooleanValue(Boolean expectedValue, String inputBytes) throws Exception {
+        assertBooleanCorrectlyParsed(true, expectedValue, inputBytes);
+        assertBooleanCorrectlyParsed(false, expectedValue, inputBytes);
+    }
+
+    /**
+     * Checks that the reader reads the expected int from the given input bits.
+     */
+    private void assertIntCorrectlyParsed(boolean constructFromBytes, int expectedValue, String inputBytes) throws Exception {
+        reader = readerForIon11(hexStringToByteArray(inputBytes), constructFromBytes);
+        assertSequence(
+            next(IonType.INT), intValue(expectedValue),
+            next(null)
+        );
+        closeAndCount();
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "                             0, 50",
+        "                             1, 51 01",
+        "                            17, 51 11",
+        "                           127, 51 7F",
+        "                           128, 52 80 00",
+        "                          5555, 52 B3 15",
+        "                         32767, 52 FF 7F",
+        "                         32768, 53 00 80 00",
+        "                        292037, 53 C5 74 04",
+        "                     321672342, 54 96 54 2C 13",
+        "                    2147483647, 54 FF FF FF 7F", // Integer.MAX_VALUE
+        "                            -1, 51 FF",
+        "                            -2, 51 FE",
+        "                           -14, 51 F2",
+        "                          -128, 51 80",
+        "                          -129, 52 7F FF",
+        "                          -944, 52 50 FC",
+        "                        -32768, 52 00 80",
+        "                        -32769, 53 FF 7F FF",
+        "                      -8388608, 53 00 00 80",
+        "                      -8388609, 54 FF FF 7F FF",
+        "                   -2147483648, 54 00 00 00 80", // Integer.MIN_VALUE
+    })
+    public void readIntValue(int expectedValue, String inputBytes) throws Exception {
+        assertIntCorrectlyParsed(true, expectedValue, inputBytes);
+        assertIntCorrectlyParsed(false, expectedValue, inputBytes);
+    }
+
+    /**
+     * Checks that the reader reads the expected long from the given input bits.
+     */
+    private void assertLongCorrectlyParsed(boolean constructFromBytes, long expectedValue, String inputBytes) throws Exception {
+        reader = readerForIon11(hexStringToByteArray(inputBytes), constructFromBytes);
+        assertSequence(
+            next(IonType.INT), longValue(expectedValue),
+            next(null)
+        );
+        closeAndCount();
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "                             0, 50",
+        "                             1, 51 01",
+        "                            17, 51 11",
+        "                           127, 51 7F",
```

Comment:
Numbers such as 127 can be encoded using more than one op code. We should add tests for that here (and in the other integer tests).
```suggestion
        "                           127, 51 7F",
        "                           127, 52 7F 00",
        "                           127, 54 7F 00 00 00",
        "                           127, 58 7F 00 00 00 00 00 00 00",
```